test/FIFOMarket.test.ts

```diff
@@ -28,48 +58,346 @@ const setupTestLocal = async (
   return {
     hre,
     contracts,
+    removal,
+    fifoMarket,
     ...rest,
+    listedRemovalIds: tokenIds,
   };
 };
 
 describe('FIFOMarket', () => {
+  describe('initialization', () => {
+    describe('roles', () => {
+      (
+        [{ role: 'DEFAULT_ADMIN_ROLE' }, { role: 'ALLOWLIST_ROLE' }] as const
+      ).forEach(({ role }) => {
+        it(`will assign the role ${role} to the deployer and set the DEFAULT_ADMIN_ROLE as the role admin`, async () => {
+          const { fifoMarket, hre } = await setupTest();
+          expect(
+            await fifoMarket.hasRole(
+              await fifoMarket[role](),
+              hre.namedAccounts.admin
+            )
+          ).to.be.true;
+          expect(await fifoMarket.getRoleAdmin(await fifoMarket[role]())).to.eq(
+            await fifoMarket.DEFAULT_ADMIN_ROLE()
+          );
+          expect(
+            await fifoMarket.getRoleMemberCount(await fifoMarket[role]())
+          ).to.eq(1);
+        });
+      });
+    });
+  });
+  describe('role access', () => {
+    describe('roles', () => {
+      describe('DEFAULT_ADMIN_ROLE', () => {
+        [
+          {
+            role: 'DEFAULT_ADMIN_ROLE',
+            accountWithRole: 'admin',
+            accountWithoutRole: 'buyer',
+          } as const,
+        ].forEach(({ role, accountWithRole, accountWithoutRole }) => {
+          it(`accounts with the role "${role}" can set the priority restricted threshold while accounts without this role cannot`, async () => {
+            const { fifoMarket, hre } = await setupTest();
+
+            const { namedAccounts, namedSigners } = hre;
+            const roleId = await fifoMarket[role]();
+            expect(
+              await fifoMarket.hasRole(roleId, namedAccounts[accountWithRole])
+            ).to.be.true;
+
+            const newThreshold = ethers.utils.parseUnits('100');
+
+            expect(
+              await fifoMarket
+                .connect(namedSigners[accountWithRole])
+                .setPriorityRestrictedThreshold(newThreshold)
+            )
+              .to.emit(fifoMarket, 'PriorityRestrictedThresholdSet')
+              .withArgs(newThreshold);
+
+            expect(await fifoMarket.priorityRestrictedThreshold()).to.equal(
+              BigNumber.from(newThreshold)
+            );
+
+            await expect(
+              fifoMarket
+                .connect(namedSigners[accountWithoutRole])
+                .setPriorityRestrictedThreshold(newThreshold)
+            ).to.be.revertedWith(
+              `AccessControl: account ${namedAccounts[
+                accountWithoutRole
+              ].toLowerCase()} is missing role ${roleId}`
+            );
+          });
+        });
+      });
+      describe('ALLOWLIST_ROLE', () => {
+        [
+          {
```

Comment:
I would either write a second test (e.g., a not allowed test) for this array and the previous one or just remove the forEach for now.